java/com/currencyone/example/Example.java

```diff
@@ -0,0 +1,103 @@
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.UUID;
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+public class Example 
+{
+    public static final String APIKEY = System.getProperty("apiKey");
+    public static final String SECRET = System.getProperty("secret");
+
+    public static void main( String[] args ) throws Exception {
+            System.out.println("Current best offers");
+            System.out.println(Example.getOrderbook());
+            System.out.println("\nMy balance");
+            System.out.println(Example.getBalance());
+            System.out.println("\nPlacing an order");
+            System.out.println(Example.placeOrder());
+    }
+
+    public static String getOrderbook() throws IOException {
+        return makeUnsignedRequest("/api/v1/public/market/orderbook/EUR_PLN");
+    }
+
+    public static String getBalance() throws Exception {
+        return makeSignedRequest("GET", "/api/v1/account/balances");
+    }
+
+    public static String placeOrder() throws Exception {
+        String uuid = UUID.randomUUID().toString();
+        return makeSignedRequest("POST", "/api/v1/market/orders"
+            + "?submitId=" + uuid 
+            + "&pair=EUR_PLN"
+            + "&price=4.231"
+            + "&buySell=BUY"
+            + "&volume=1.00"
+            + "&volumeCurrency=EUR"
+            + "&otherCurrency=PLN");
+    }
+
+    public static String makeUnsignedRequest(String uri) throws IOException {
+        URL url = new URL("https://api.walutomat.pl" + uri);
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+        return connectionResponseToString(connection);
+    }
+
+    public static String makeSignedRequest(String method, String uri) throws Exception {
+        Long ts = System.currentTimeMillis();
+        String signature = encode(SECRET, uri + ts);
+        URL url = new URL("https://api.walutomat.pl" + uri);
+
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+        connection.setRequestMethod(method);
+        connection.setRequestProperty("X-API-KEY", APIKEY);
+        connection.setRequestProperty("X-API-NONCE", ts.toString());
+        connection.setRequestProperty("X-API-SIGNATURE", signature);
+        return connectionResponseToString(connection);
+    }
+
+    public static String connectionResponseToString(HttpURLConnection connection) throws IOException {
+        try (InputStream is = connection.getInputStream()) {
+            return inputStreamToString(is); 
+        } catch (IOException e) {
+            try (InputStream is = connection.getErrorStream()) {
+                throw new IllegalArgumentException(inputStreamToString(is));
+            }
+        }
+    }
+
+    public static String inputStreamToString(InputStream is) {
+        java.util.Scanner s = new java.util.Scanner(is, "UTF8").useDelimiter("\\A");
```

Comment:
Few lines below there is `UTF-8`, here it's `UTF8`, both are OK? Would normalize, I guess the `UTF-8` is the correct one.